packages/fx-core/src/plugins/resource/bot/errors.ts

```diff
@@ -14,6 +14,22 @@ export enum ErrorType {
   System,
 }
 
+function InferInnerError(target: PluginError, helpLinkMap: Map<string, string>): void {
+  if (!target.innerError) return;
+
+  const errorCode = target.innerError.response?.data?.error?.code;
+  const helpLink = helpLinkMap.get(errorCode);
+  if (helpLink) target.helpLink = helpLink;
+
+  const statusCode = target.innerError.response?.status;
+  if (!statusCode) return;
+  if (statusCode >= Constants.statusCodeUserError && statusCode < Constants.statusCodeServerError) {
```

Comment:
NIT: I don't think 400 and 500 are magic numbers here, and I prefer use them directly in code.
`statusCode >= 400 && statusCode < 500 `, which is more readable for me.

Reply:
Both ways are fine to me, so I would leave it to Ivan to decide.